cpp/src/Slice/Parser.h

```diff
@@ -1230,8 +1235,23 @@ class Unit : public virtual Container
 
     int parse(const std::string&, FILE*, bool);
 
-    virtual void destroy();
-    virtual void visit(ParserVisitor*, bool);
+    void destroy() override;
+    ContainedList contents() const override;
+    void visit(ParserVisitor*, bool) override;
+    bool hasSequences() const;
+    bool hasStructs() const;
+    bool hasExceptions() const;
+    bool hasDictionaries() const;
```

Comment:
It would be good to eliminate functions that are not used anywhere (such as `hasDictionaries`).

Reply:
It's also likely all these functions - that are called rarely, sometimes just once - are not always called on both Unit and Module.

Reply:
Oh no, there's definitely more than a couple that either aren't called on both, and that aren't even called anywhere. I left them because they seemed like generally useful utility methods, but in hindsight, yeah, if anyone needs them, I guess they can rewrite them themselves.
Removed all the ones that aren't used.